### ISSUE TITLE
ERROR: I/O error while communicating with vault

### ISSUE THREAD

**Issue by rockyr1** [Nike-Inc/cerberus-doc-site#34]
Followed all the steps to setup the Vault & consul infrastrcuture. We are stuck with below error when we try to initialize vault. Has anyone come across this issue? 

`ERROR: I/O error while communicating with vault`

Have rebuilt the entire stack thrice now. Still no luck. 

**Consul Log**
`2017/01/24 19:45:38 [ERR] agent: coordinate update error: rpc error: failed to get conn: remote error: bad certificate`

**Vault log**
```
Error initializing core: missing advertisement address
Error detecting advertise address: Get https://127.0.0.1:8500/v1/agent/self?token=12345-678-123-1234: remote error: bad certificate
Error initializing core: missing advertisement address
Error detecting advertise address: Get https://127.0.0.1:8500/v1/agent/self?token=12345-678-123-1234: remote error: bad certificate
Error initializing core: missing advertisement address

```

**Comment by tlisonbee** [comment]
I've encountered the I/O error before due to network problems between my machine and the vault instances (not having my IP whitelisted, corporate firewall, socks proxy settings, etc).

Also, when I tested the "creating an environment" instructions a few weeks ago, the other issue I hit was not having certificates setup correctly.  The error message above "bad certificate" makes me wonder if that is what you are hitting.

**Comment by fieldju** [comment]
I am suspecting that you are experiencing issues related to a bad certificate, ca or private key.

Please do read the following articles
https://www.consul.io/docs/agent/encryption.html
from https://www.consul.io/docs/agent/options.html

cert_file This provides a file path to a PEM-encoded certificate. The certificate is provided to clients or servers to verify the agent's authenticity. It must be provided along with key_file.

key_file This provides a the file path to a PEM-encoded private key. The key is used with the certificate to verify the agent's authenticity. This must be provided along with cert_file.

ca_file This provides a file path to a PEM-encoded certificate authority. The certificate authority is used to check the authenticity of client and server connections with the appropriate verify_incoming or verify_outgoing flags.

How did you create the certificates?

**Comment by rockyr1** [comment]
@fieldju We generated cert using venafi. With pfx, we extracted cert, key and chain.

@tlisonbee Yes, thats kind of what i am thinking as well. We ran into bad certificate issue, we will regenerate the certs and try. 

Thank you both for your inputs, will test and report back.

**Comment by tlisonbee** [comment]
@rockyr1 how did it go?

I looked back at some notes on using the Venafi UI.  We used "Entrust Root Certification Authority" and when you download the Certificate, you want to choose PEM/OpenSSL and check all of the checkboxes (Root chain, Private key).

Apologies, I know this is the most error prone part of setting up a Cerberus environment.  Ideally we'd have it completely automated with Let'sEncrypt so that it would be super easy to get an environment up to try it out (but our team won't have the chance to get to that at least for the next few months).
